src/library/Flex/Flex.js

```diff
@@ -63,60 +66,123 @@ export type Values = boolean | null | number | string;
 const getAlignment = (value: string): string =>
   ['start', 'end'].indexOf(value) !== -1 ? `flex-${value}` : value;
 
-const getGutterSize = (theme: Object, value: number | string): string =>
-  typeof value === 'number'
-    ? pxToEm(value / 2)
-    : `${parseFloat(theme[`space_inline_${value}`] || value) / 2}em`;
-
-const getMargin = (
-  gutter: string,
+const getGutterSize = (
   theme: Object,
   value: number | string
-): string =>
-  value ? `calc(${getMeasurement(theme, value)} - ${gutter})` : `-${gutter}`;
+): number | string =>
+  typeof value === 'number' && value !== 0
+    ? pxToEm(value)
+    : theme[`space_inline_${value}`] || value;
 
-const getMeasurement = (theme: Object, value: number | string): string => {
-  return (
-    theme[`space_inline_${value}`] ||
-    (typeof value === 'number' ? `${value}px` : value)
-  );
-};
+const getIndexedValue = (property, index) =>
+  property && Array.isArray(property) && index !== undefined
+    ? property[index]
+    : property;
 
 const getJustification = (value: string): string =>
   ['around', 'between', 'evenly'].indexOf(value) !== -1
     ? `space-${value}`
     : getAlignment(value);
 
+const getMarginOrGutter = ({
+  gutterWidth,
+  index,
+  margin,
+  marginEnd,
+  marginHorizontal,
+  marginStart,
+  start,
+  theme
+}) => {
+  return (
+    getIndexedValue(start ? marginStart : marginEnd, index) ||
+    getIndexedValue(marginHorizontal, index) ||
+    getIndexedValue(margin, index) ||
+    getGutterSize(theme, gutterWidth)
+  );
+};
+
+const getMarginProps = ({ direction, gutterWidth, theme, ...restProps }) => {
+  if (Array.isArray(direction)) {
+    return direction.reduce(
+      (acc, _, index) => {
+        const value = getPrevNonNull(direction, index);
+
+        if (value === 'row' || value === 'row-reverse') {
+          pushMarginProps({
+            direction: value,
+            index,
+            gutterWidth,
+            props: acc,
+            theme,
+            ...restProps
+          });
+        } else {
+          pushMarginProps({
+            index,
+            gutterWidth: 0,
+            props: acc,
+            theme,
+            ...restProps
+          });
+        }
+        return acc;
+      },
+      { marginStart: [], marginEnd: [] }
+    );
+  } else if (direction === 'row' || direction === 'row-reverse') {
+    const flip = direction === 'row-reverse';
+    const marginProperty = flip ? 'marginStart' : 'marginEnd';
+
+    return {
+      [marginProperty]: getMarginOrGutter({
+        ...restProps,
+        gutterWidth,
+        start: flip,
+        theme
+      })
+    };
+  }
+};
+
+const pushMarginProps = ({
+  direction,
+  index,
+  gutterWidth,
+  props,
+  theme,
+  ...restProps
+}) => {
+  const flip = direction === 'row-reverse';
```

Comment:
This could be pulled out to decrease duplication; small change though, not critical at all

Reply:
The `direction` referenced here is a value in an array, whereas the other declaration of `flip` uses the non-array, string value, so the best I could do is make it a function, which seems overkill.

I briefly considered it anyway, as it would be nice to have that `'row-reverse'` detail in one spot, but quickly realized I'd need to refactor much of the file to be consistent in such an approach, as I regular compare against "special strings" like that. So I abandoned that. 😅